src/hb-array.hh

```diff
@@ -173,16 +173,25 @@ struct hb_array_t : hb_iter_with_fallback_t<hb_array_t<Type>, Type&>
 
   void reverse ()
   {
-    int rhs = length - 1;
-    int lhs = 0;
-    while (rhs > lhs)
-    {
-      Type value_rhs = arrayZ[rhs];
-      Type value_lhs = arrayZ[lhs];
-      arrayZ[rhs] = value_lhs;
-      arrayZ[lhs] = value_rhs;
-      rhs--;
-      lhs++;
+    return reverse (0, length);
```

Comment:
Make start/end have defaults so you don't need separate override.

Reply:
Done.